lib/addict/interactors/addict_manager_interactor.ex

```diff
@@ -104,13 +111,13 @@ defmodule Addict.BaseManagerInteractor do
         throw "Unable to create user, invalid hash: nil"
       end
 
-      defp validate_params(user_params) do
-        case is_nil(user_params["email"])
-             or is_nil(user_params["password"])
-             or is_nil(user_params["username"]) do
-               false -> user_params
-               true -> throw "Unable to create user, invalid hash. Required params: email, password, username"
-             end
+      # With default fields to validate or with option to override
+      defp validate_params(user_params, required \\ ["email","password","username"]) do
+        empty = Enum.count(required, fn(key) -> is_nil(user_params[key]) end)
+        case empty do
+          0 -> {:ok, user_params}
+          _ -> {:error, "Required parameters are missing"}
```

Comment:
Can you inject the `required` param on the exception message please? It might be really helpful to the developer.